src/test/java/uk/ac/ebi/ena/webin/cli/manifest/processor/metadata/SampleProcessorTest.java

```diff
@@ -12,43 +12,218 @@
+        Assert.assertFalse(result.isValid());
+        Assert.assertEquals("SRP000392", fieldValue.getValue());
+    }
+
+
+    /**
+     * Test Add and Modify with setSampleUpdate parameter.
+     * The collectionDate in sample object will be updated to current date. 
+     **/
+    @Test
+    public void
+    testJsonSampleAddAndUpdate() throws JsonProcessingException {
         
-        ManifestFieldValue fieldValue = createFieldValue( ManifestFieldType.META, "SAMPLE", "ERS000002" );
+        String sampleTestAlias = UUID.randomUUID().toString();
+       
+        // Create a new sample
+        SampleProcessor processor = new SampleProcessor(parameters,
+                (Sample sample) -> {
+                    Assert.assertNotNull(sample.getBioSampleId());
+                    Assert.assertEquals(getSampleCollectionDate(sample), "2010-01-20");
+                });
+        ManifestFieldValue fieldValue = createFieldValue(ManifestFieldType.META, "SAMPLE", getSampleJson(sampleTestAlias));
         ValidationResult result = new ValidationResult();
-        processor.process( result, fieldValue );
-        Assert.assertTrue(result.isValid() );
-        Assert.assertEquals( "SAMEA749881", fieldValue.getValue() );
+        processor.process(result, fieldValue);
+        Assert.assertTrue(result.isValid());
+        Assert.assertNotNull(fieldValue.getValue().startsWith("SAMEA"));
+        String biosampleId = fieldValue.getValue();
+        
+        // Update sample
+        JsonNode sampleJson = new ObjectMapper().readTree(getSampleJson(sampleTestAlias));
+        // Update collection date to current date
+        sampleJson.get("attributes").forEach(e ->{
+            if(e.get("tag").asText().equals("collection date")){
+                ((ObjectNode)e).put("value", LocalDate.now().toString());
+            }
+        });
+        processor = new SampleProcessor(parameters,
+                (Sample sample) -> {
+                    // Updated collection date
+                    Assert.assertEquals( getSampleCollectionDate(sample),(LocalDate.now().toString()));
+                }
+        );
+        parameters.setSampleUpdate(true);
+
+        Sample sampleObj = testGetSampleUsingValidId(sampleTestAlias);
+        Assert.assertEquals(getSampleCollectionDate(sampleObj), "2010-01-20");
+        fieldValue = createFieldValue(ManifestFieldType.META, "SAMPLE", sampleJson.toString());
+        result = new ValidationResult();
+        processor.process(result, fieldValue);
+        Assert.assertTrue(result.isValid());
+        Assert.assertEquals(biosampleId, fieldValue.getValue());
     }
 
-    
-    @Test public void 
-    testIncorrect()
-    {
-        SampleProcessor processor = new SampleProcessor( parameters, Assert::assertNull );
-        ManifestFieldValue fieldValue = createFieldValue( ManifestFieldType.META, "SAMPLE", "SRP000392" );
+
+    /**
+     * Test Add and Modify without setSampleUpdate parameter.
```

Comment:
Test add and modify sample submission without -setSampleUpdate option. We change the collection date and check that the sample does not get updated.